javascript/javascript3/week3/shopping cart/app.js

```diff
@@ -0,0 +1,95 @@
+// Shopping cart using Classes
+const body = document.querySelector("body");
+
+class Product {
+    constructor(name, price) {
+        this.name = name;
+        this.price = price;
+    }
+    convertToCurrency(currency) {
+        let convertedPrice = this.price;
+        switch (currency) {
+            case "dollars":
+                convertedPrice = this.price * 0.15;
+                break;
+            case "euros":
+                convertedPrice = this.price * 0.13;
+                break;
+            case "NOK":
+                convertedPrice = this.price * 1.32;
+                break;
+        }
+        return convertedPrice;
+    }
+}
+
+class ShoppingCart {
+    constructor() {
+        this.products = [];
+    }
+
+    addProduct(product) {
+        this.products.push(product);
+    }
+
+    removeProduct(product) {
+        this.products.splice(this.products.indexOf(product), 1);
+    }
+
+    searchProduct(productName) {
+        let searchedProduct = {};
+        this.products.forEach((product) => {
+            if (product.name === productName) {
+                searchedProduct = product;
+            }
+        });
+        return searchedProduct;
+    }
+
+    getTotal() {
+        let sum = 0;
+        for (let i = 0; i < this.products.length; i++) {
+            sum += this.products[i].price;
+        }
+        return sum;
+    }
```

Comment:
Using "reduce" is another good way.

 const sum= this.products.reduce((prev, next) => prev + next.price, 0);